### PR TITLE
Parse let declarations within block items

### PR DESCRIPTION
**Pull request by evanbryantmiller**

### CODE REVIEW

**Comment by evanbryantmiller** [comment]
I initially misunderstood the syntax.  This implementation covers all let declarations.

File: tools/syntax.txt [hunk 1]
```diff
@@ -982,6 +985,24 @@ list<SyntaxNode> items
 token end
 NamedBlockClauseSyntax endBlockName
 
+LetPortSyntax kind=LetPort
+list<AttributeInstanceSyntax> attributes
+DataTypeSyntax type
+token identifier
```
Comment (MikePopoloski) [inline tools/syntax.txt]: You could probably reuse VariableDeclaratorSyntax for the following three entries and save some work.

File: tools/syntax.txt [hunk 2]
```diff
@@ -982,6 +985,24 @@ list<SyntaxNode> items
...
+LetPortListSyntax kind=LetPortList
+token openParen
+separated_list<LetPortSyntax> ports
+token closeParen
+
+LetDeclarationSyntax kind=LetDeclaration base=MemberSyntax
+token let
+token identifier
+LetPortListSyntax portList
+EqualsValueClauseSyntax initializer
```
Comment (MikePopoloski) [inline tools/syntax.txt]: Missing semicolon?

File: source/Parser.cpp [hunk 3]
```diff
@@ -1653,6 +1653,41 @@ MemberSyntax* Parser::parseVariableDeclaration(ArrayRef<AttributeInstanceSyntax*
         return alloc.emplace<ParameterDeclarationStatementSyntax>(attributes, parameter, semi);
     }
 
+    if (peek(TokenKind::LetKeyword)) {
+        auto let = consume();
+        auto identifier = expect(TokenKind::Identifier);
+        auto openParen = expect(TokenKind::OpenParenthesis);
```
Comment (MikePopoloski) [inline source/Parser.cpp]: It looks like the port list is optional.

**Pull request merged**